torch/standalone/slim_tensor/array_ref.h

```diff
@@ -0,0 +1,180 @@
+#pragma once
+
+#include <torch/standalone/slim_tensor/shared_ptr.h>
+#include <array>
+#include <cassert>
+#include <cstdint>
+#include <cstring>
+#include <vector>
+
+namespace torch::standalone {
```

Comment:
Do we have a design document somewhere on what should be in `torch::` vs `at::` namespace? Because this feels like it belongs to `at::` namespace rather than torch one